### PR TITLE
[feature] Poll web view

### PR DESCRIPTION
**Pull request by tsmethurst**
# Description

> If this is a code change, please include a summary of what you've coded, and link to the issue(s) it closes/implements.
>
> If this is a documentation change, please briefly describe what you've changed and why.

This pull request adds a template + web view for rendering status polls in the frontend.

Produced HTML for a poll looks like this:

```html
<figure class="poll">
	<figcaption class="poll-info">
		<span class="poll-expiry">Poll closed&nbsp;Aug 28, 2021, 10:40</span>
		<span class="total-votes">Total votes: 2</span>
	</figcaption>
	<ul class="poll-options">
		<li class="poll-option">
			<label aria-hidden="true" for="option-1" lang="en">50:50</label>
			<meter aria-hidden="true" id="option-1" min="0" max="100" value="0">0&#37;</meter>
			<div class="sr-only">Option 1:&nbsp;<span lang="en">50:50</span></div>
			<div class="poll-vote-summary">
				<span class="poll-vote-share">0&#37;</span>
				<span class="poll-vote-count">0 votes</span>
			</div>
		</li>
		<li class="poll-option">
			<label aria-hidden="true" for="option-2" lang="en">phone a friend</label>
			<meter aria-hidden="true" id="option-2" min="0" max="100" value="50">50&#37;</meter>
			<div class="sr-only">Option 2:&nbsp;<span lang="en">phone a friend</span></div>
			<div class="poll-vote-summary">
				<span class="poll-vote-share">50&#37;</span>
				<span class="poll-vote-count">1 vote</span>
			</div>
		</li>
		<li class="poll-option">
			<label aria-hidden="true" for="option-3" lang="en">ask the audience</label>
			<meter aria-hidden="true" id="option-3" min="0" max="100" value="50">50&#37;</meter>
			<div class="sr-only">Option 3:&nbsp;<span lang="en">ask the audience</span></div>
			<div class="poll-vote-summary">
				<span class="poll-vote-share">50&#37;</span>
				<span class="poll-vote-count">1 vote</span>
			</div>
		</li>
	</ul>
</figure>
```

And it's styled in the frontend like this:

image

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [ ] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/typeutils/internaltofrontend.go [hunk 1]
```diff
@@ -678,6 +678,48 @@ func (c *Converter) StatusToWebStatus(
...
+			// is distinct from "1.68%" in polls with loads
+			// of votes.
+			//
+			// However, if we've got eg., a two-option poll
+			// in which each option has half the votes, then
+			// "50%" looks better than "50.00%".
+			//
+			// By the same token, it's pointless to show
+			// "0.00%" or "100.00%".
+			voteShareStr := fmt.Sprintf("%.2f", voteShare)
```
Comment (NyaaaWhatsUpDoc) [inline internal/typeutils/internaltofrontend.go:707]: ooh i like this way of doing this. i think my method would have been some horrible maths but this makes much more sense in a quick-and-easy-to-see way

File: internal/typeutils/internaltofrontend.go [hunk 2]
```diff
@@ -1456,10 +1498,17 @@ func (c *Converter) PollToAPIPoll(ctx context.Context, requester *gtsmodel.Accou
...
 
-	// TODO: emojis used in poll options.
-	// For now init to empty slice to serialize as `[]`.
-	// In future inherit from parent status.
-	emojis = make([]apimodel.Emoji, 0)
+	// Try to inherit emojis
+	// from parent status.
+	if pStatus := poll.Status; pStatus != nil {
+		var err error
+		emojis, err = c.convertEmojisToAPIEmojis(ctx, pStatus.Emojis, pStatus.EmojiIDs)
```
Comment (NyaaaWhatsUpDoc) [inline internal/typeutils/internaltofrontend.go:1505]: in time we should filter these for those specifically in use by the poll options, not hugely important but would help save some bandwidth :]

**Comment by NyaaaWhatsUpDoc** [comment]
looks good to me! squerge away!!

File: internal/typeutils/internaltofrontend.go (hunk 2 above)
Comment (tsmethurst) [reply internal/typeutils/internaltofrontend.go:1505]: Yeah I was thinking that too, this isn't ideal but it's good enough for now :)

**Pull request merged**